src/stdlib_stats_distribution_PRNG.fypp

```diff
@@ -0,0 +1,211 @@
+#:include "common.fypp"
+module stdlib_stats_distribution_PRNG
+    use stdlib_kinds
+    implicit none
+    private
+    integer, parameter :: MAX_INT_BIT_SIZE = 64
+    integer(int64), save :: st(4), si = 614872703977525537_int64
+    logical, save :: seed_initialized = .false.
+
+    public :: random_seed
+    public :: dist_rand
+    public :: jump
+    public :: long_jump
+
+
+    interface dist_rand
+    !! Version experimental
+    !!
+    !! Generation of random integers with different kinds
+    !! ([Specification](../page/specs/stdlib_stats_distribution_PRNG.html#
+    !! description))
+        #:for k1, t1 in INT_KINDS_TYPES
+        module procedure dist_rand_${t1[0]}$${k1}$
+        #:endfor
+    end interface dist_rand
+
+    interface random_seed
+    !! Version experimental
+    !!
+    !! Set seed value for random number generator
+    !! ([Specification](../page/specs/stdlib_stats_distribution_PRNG.html#
+    !! description))
+    !!
+        #:for k1, t1 in INT_KINDS_TYPES
+        module procedure random_distribution_seed_${t1[0]}$${k1}$
+        #:endfor
+    end interface random_seed
+
+
+    contains
+
+    #:for k1, t1 in INT_KINDS_TYPES
+    function dist_rand_${t1[0]}$${k1}$(n) result(res)
+    !! Random integer generation for various kinds
+    !! result = [-2^k, 2^k - 1], k = 7, 15, 31, 63, depending on input kind
+    !! Result is used as bit model number instead of regular arithmetic number
+    !!
+        ${t1}$, intent(in) :: n
+        ${t1}$ :: res
+        integer :: k
+
+        k = MAX_INT_BIT_SIZE - bit_size(n)
+        res = shiftr(xoshiro256ss( ), k)
+    end function dist_rand_${t1[0]}$${k1}$
+
+    #:endfor
+
+    function xoshiro256ss( ) result (res)
+    ! Generate random 64-bit integers
+    ! Written in 2018 by David Blackman and Sebastiano Vigna (vigna@acm.org)
```

Comment:
Please mention a complete reference (link,...) as well as the license. Is it compatible with the MIT license of `stdlib`?

Reply:
It is in public domain authorized by the owners. [http://prng.di.unimi.it/xoshiro256starstar.c]

Reply:
Thank you for the information. Could you also add it this info and link in the code to document it, please?

Reply:
OK